plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2287,13 +2287,28 @@ public void validateScan(ConnectorSession session, ConnectorTableHandle tableHan
         Map<ConnectorExpression, ProjectedColumnRepresentation> columnProjections = projectedExpressions.stream()
                 .collect(toImmutableMap(Function.identity(), HiveApplyProjectionUtil::createProjectedColumnRepresentation));
 
-        // No pushdown required if all references are simple variables
+        HiveTableHandle hiveTableHandle = (HiveTableHandle) handle;
+        // all references are simple variables
         if (columnProjections.values().stream().allMatch(ProjectedColumnRepresentation::isVariable)) {
-            return Optional.empty();
+            Set<ColumnHandle> projectedColumns = ImmutableSet.copyOf(assignments.values());
+            if (hiveTableHandle.getProjectedColumns().isPresent()
+                    && hiveTableHandle.getProjectedColumns().get().containsAll(projectedColumns)) {
+                return Optional.empty();
```

Comment:
Why empty? I think this should be `hiveTableHandle.getProjectedColumns().get().equals(projectedColumns)`. Otherwise, we are pruning columns further. Please add test.

Reply:
It should be equals, fixing
Empty because we don't need to do anything more